base/exports.jl

```diff
@@ -1430,6 +1430,7 @@ export
     @inbounds,
     @fastmath,
     @simd,
+    @shareindexes,
```

Comment:
Maybe make it more general? `@sharediterator`? There's nothing inherently index-y about this macro right now...